feat: remove-jquery

alexsoin opened the pull request:
Удаление зависимости от jQuery

- переписал код vanilla js
- добавил стили для модального окна
- удалил плагин `jquery.magnific-popup`
- удалил неиспользуемые стили `.mfp-*`

alexsoin committed:
feat: remove-jquery

alexsoin committed:
feat: update ui

- Замена тега получения manager_url на fenom
- Обновление вида панели
- Отключения перехода по элементам панели через tab

webinmd commented:
если несколько раз открыть/закрыть окно редактирования, в консоли появляются js ошибки. Они не влияют на работу компонента, но явно лишние

![image](https://github.com/modx-pro/frontendmanager/assets/966234/c2c9e4e6-dd76-4112-a799-0dfc38cd24c9)

webinmd commented:
Если сайт на английском, то слово Загрузка остается на русском языке, хотя все пункты в панельке меняются на английский. Можно в принципе на какую-то loading анимацию заменить, если это вшито в js файл

alexsoin committed:
fix: bugs

Добавление функционала и лексикона для вывода сообщения о загрузке фрейма
Исправлена ошибка 'removeChild' при закрытии модального окна
Вынесены классы и id элементов в конфиг
Убрана обводка у фрейма

alexsoin committed:
fix: loader text

alexsoin committed:
fix: modal window behavior

GulomovCreative committed:
feat: Улучшение взаимодействия с элементами панели

alexsoin committed:
Merge pull request #1 from GulomovCreative/remove-jquery

feat: Улучшение взаимодействия с элементами панели

alexsoin committed:
fix: Исправлена ошибка при нажатии кнопки закрыть

OlegAnTo2000 committed:
Увеличить модальное окно, пофиксить размытие

Из-за выравнивания модального окна по центру экрана в помощью transition на некоторых устройствах содержимое iframe размывается. Кроме того, удобнее работать с большим модальным окном, оставив минимальные поля для закрытия. Сделал фиксированный размер кнопки закрытия, выравнивание модального окна в центре путем указания top, right, bottom и left.

alexsoin committed:
Merge pull request #2 from OlegAnTo2000/patch-1

Увеличить модальное окно, пофиксить размытие

alexsoin committed:
feat: rocket update 🚀

- Для единообразия отступы в стилях сконвертированы в табы
- Переписан вывод модального окна: добавлен padding в iframe, переписан адаптив, добавлена максимальная ширина модалки
- Переписано подключение ресурсов и вывод чанка, чтобы теги не добавлялись за пределы body
- Добавлен paddingRight для предотвращения дерганья контента при открытии модального окна #19
- Устранено подергивание панели при наличии класса `fm-hide` при обновлении страницы #20
- Добавление defer #17

alexsoin committed:
fix: box-sizing

alexsoin committed:
fix: modal box-sizing
